### DISCUSSION TITLE
Persian calendar to compute NextFireTime in triggers

### DISCUSSION THREAD

**Discussion by sadegh-alinejad** [quartznet/quartznet#2319] [Q&A]
Hi
We want to use Quartz 3.8.1 in our system with two types of Simple and Cron scheduling, but we have different calendar in our system than Gregorian calendar which is "Persian calendar". Our problem is in Persian calendar first six month of year is 31 days and the second 6 month of year is 30 days, so if we use Corn based calendar for monthly jobs, we will face different time we need.
First I thought if I use custom calendar (a class that overrides BaseCalendar) and assign it to my trigger it will call GetNextIncludedTimeUtc method for computing next month, but its not. Seems it is working only for excluding time. I read the source code, for example SimpleTriggerImpl.cs and CalendarIntervalTriggerImpl.cs are using their own logic to compute next fire time.

If I am not missing something, I really appreciate that to give me a solution to customize computing next fire time.

Thanks a lot.

**Comment by lahma** [comment]
I'm afraid that you need to implement the triggers yourself to work with Persian calendar system. There's no capacity to support such work, not sure if there's any scheduling library capable of handling other than Gregorian calendar.

**Reply by sadegh-alinejad** [reply]
Okay, thanks. Do you have any plans to add this support in future? if not I think we should create our own trigger

**Reply by lahma** [reply]
I don't have such plans as I don't have a need for that kind of feature.
